### PR TITLE
Add: tee(), copy into container during iteration

### PR DESCRIPTION
**Pull request by Kijewski**
This change also moves the inside of sink() into its own function: sink_one()

### CODE REVIEW

File: include/rx/ranges.hpp [hunk 1]
```diff
@@ -1998,6 +2001,58 @@ struct reverse {
...
+            if (!sunk) {
+                sink_one(input.get(), dest);
+            }
+            input.next();
+            sunk = false;
+        }
+
+        constexpr output_type get() const {
+            auto&& value = input.get();
+            sink_one(value, dest);
```
Comment (simonask) [inline include/rx/ranges.hpp]: Why sink during `get()`?

It seems like sinking during `next()` should be good enough, and do away with the `mutable bool sunk` member.

**Comment by simonask** [comment]
Feel free to add an entry to `CHANGELOG.md` as well under "Features". :-)

**Review by Hazurl — APPROVED** [review]

**Review by simonask — APPROVED** [review]

**Comment by simonask** [comment]
@Kijewski Would you mind merging `version-2` into this branch (again)? The conflicts are bit too much for Github's merge editor. 😅

**Comment by Kijewski** [comment]
I rebases the changes. (I'd I re-did the commits. Using git mergetool was to complicated is this case).

**Review by simonask — APPROVED** [review]

**Pull request merged**